File: JavaScript/6js 1/fileNameExtractor.js
/*
|--------------------------------------------------------------------------
| extract portion of file name
|--------------------------------------------------------------------------
|
*/

class FileNameExtractor {
   static extractFileName(dirtyFileName) {
      const regex = /^[0-9]+_(.+?\.[a-zA-Z0-9]+)(?:\.[a-zA-Z0-9]+)+$/;
      const match = dirtyFileName.match(regex);
      return match ? match[1] : null;
   }
}


class FileNameExtractor {
   static extractFileName = (dirty) => dirty.match(/^\d+_([^.]+\.[^.]+)/)[1];
}


class FileNameExtractor {
   static extractFileName(dFN) {
      return dFN.slice(dFN.indexOf("_") + 1, dFN.lastIndexOf("."));
   }
}

File: Python/6py 1/fileNameExtractor.py
"""
|--------------------------------------------------------------------------
| extract portion of file name
|--------------------------------------------------------------------------
|
"""

from re import search
class FileNameExtractor:
   @staticmethod
   def extract_file_name(dirty_file_name):
      match = search(r'_(.*)\.', dirty_file_name)
      if match: return match.group(1)
      return None


class FileNameExtractor:
   @staticmethod
   def extract_file_name(fname):
      return fname.split('_', 1)[1].rsplit('.', 1)[0]
   

from re import match
class FileNameExtractor:
   def extract_file_name(dirty_file_name):
      return match(r'\d+_(.+?\..+?)\.', dirty_file_name).group(1)